File: backend/app/schemas/mensajes.py
from __future__ import annotations

from datetime import datetime
from uuid import UUID

from pydantic import BaseModel, ConfigDict


class MensajeCreate(BaseModel):
    model_config = ConfigDict(extra="forbid")

    asunto: str
    cuerpo: str
    destinatario_id: UUID
    hilo_id: UUID | None = None


class MensajeReply(BaseModel):
    model_config = ConfigDict(extra="forbid")

    asunto: str
    cuerpo: str


class MensajeResponse(BaseModel):
    model_config = ConfigDict(extra="forbid")

    id: UUID
    tenant_id: UUID
    asunto: str
    cuerpo: str
    remitente_id: UUID
    destinatario_id: UUID
    hilo_id: UUID
    padre_id: UUID | None = None
    leido_at: datetime | None = None
    created_at: datetime
    updated_at: datetime


class InboxThreadItem(BaseModel):
    model_config = ConfigDict(extra="forbid")

    hilo_id: UUID
    remitente_id: UUID
    destinatario_id: UUID
    ultimo_asunto: str
    ultimo_cuerpo: str
    leido: bool
    ultima_actividad: datetime

File: frontend/src/features/mensajes/types/mensajes.ts
export interface InboxThreadItem {
  hilo_id: string;
  remitente_id: string;
  destinatario_id: string;
  ultimo_asunto: string;
  ultimo_cuerpo: string;
  leido: boolean;
  ultima_actividad: string;
}

export interface MensajeCreate {
  asunto: string;
  cuerpo: string;
  destinatario_id: string;
  hilo_id?: string;
}

export interface MensajeReply {
  asunto: string;
  cuerpo: string;
}

export interface MensajeResponse {
  id: string;
  tenant_id: string;
  asunto: string;
  cuerpo: string;
  remitente_id: string;
  destinatario_id: string;
  hilo_id: string;
  padre_id?: string;
  leido_at?: string;
  created_at: string;
  updated_at: string;
}
